Sources/TokamakDOM/Views/Canvas/Canvas.swift

```diff
@@ -0,0 +1,514 @@
+      _ = canvasContext.quadraticCurveTo!(
+        Double(rect.rect.minX),
+        Double(rect.rect.minY),
+        Double(rect.rect.minX + cornerSize.width),
+        Double(rect.rect.minY)
+      )
+    case let .path(box):
+      for element in box.elements {
+        switch element {
+        case let .move(point):
+          _ = canvasContext.moveTo!(Double(point.x), Double(point.y))
+        case let .line(point):
+          _ = canvasContext.lineTo!(Double(point.x), Double(point.y))
+        case let .quadCurve(endPoint, controlPoint):
+          _ = canvasContext.quadraticCurveTo!(
+            Double(controlPoint.x),
+            Double(controlPoint.y),
+            Double(endPoint.x),
+            Double(endPoint.y)
+          )
+        case let .curve(endPoint, control1, control2):
+          _ = canvasContext.bezierCurveTo!(
+            Double(control1.x),
+            Double(control1.y),
+            Double(control2.x),
+            Double(control2.y),
+            Double(endPoint.x),
+            Double(endPoint.y)
+          )
+        case .closeSubpath:
+          _ = canvasContext.closePath!()
+          _ = canvasContext.beginPath!()
+        }
+      }
+    default:
+      print("TODO: push \(path) (\(path.storage))")
+    }
+    _ = canvasContext.closePath!()
+  }
+
+  private func applyModifiers(
+    _ modifiers: [Text._Modifier],
+    to canvas: JSObject,
+    in environment: EnvironmentValues
+  ) {
+    var style = ""
+    var variant = ""
+    var weight = ""
+    var size: String?
+    var lineHeight = ""
+    var family: String?
+    for modifier in modifiers {
+      switch modifier {
+      case let .color(color):
+        if let color = color {
+          canvas.fillStyle = GraphicsContext.Shading.color(color)
+            .cssValue(in: environment, with: canvas, bounds: .zero)
+        }
+      case let .font(font):
+        if let font = font {
+          let styles = font.styles(in: environment)
+          style += styles["font-style"] ?? ""
+          variant += styles["font-variant"] ?? ""
+          weight = styles["font-weight"] ?? ""
+          size = styles["font-size"] ?? ""
+          lineHeight = styles["line-height"] ?? ""
+          family = styles["font-family"] ?? ""
+        }
+      case .italic:
+        style += "italic"
+      case let .weight(w):
+        if let value = w?.value {
+          weight = "\(value)"
+        }
+      case .kerning, .tracking, .rounded, .baseline, .strikethrough,
+           .underline: break // Not supported in <canvas>.
+      }
+    }
+
+    canvas
+      .font =
+      .string(
+        "\(style) \(variant) \(weight) \(size ?? "17")pt \(lineHeight) \(family ?? Font.Design.default.families.joined(separator: " "))"
+      )
+  }
+}
+
+extension Canvas: DOMPrimitive {
+  public var renderedBody: AnyView {
+    AnyView(_Canvas(parent: self))
+  }
+}
+
+private extension GraphicsContext.Shading {
```

Comment:
Extensions below this line could into a separate file? Yes, they would no longer stay `private`, but it would potentially improve readability and overall navigation through the project, preventing files from becoming larger.